### ISSUE TITLE
non site based reporting

### ISSUE THREAD

**Issue by drfaustusfade** [drfaustusfade/ngm-reportDesk#58]
to check if we can have a location without site_id and site_name

**Comment by drfaustusfade** [comment]
to check if validation can be set per country

**Comment by drfaustusfade** [comment]
to check if titles without site_name will be displayed correctly

**Comment by fakhrihawari** [comment]
> to check if titles without site_name will be displayed correctly

<img width="1162" alt="Screen Shot 2021-02-18 at 09 25 54" src="https://user-images.githubusercontent.com/12655589/108296017-55fee500-71cb-11eb-8100-155d666bfe7d.png">

Just Like this?

**Comment by fakhrihawari** [comment]
> to check if validation can be set per country

I think we can do that

**Comment by fakhrihawari** [comment]
> to check if we can have a location without site_id and site_name



> to check if we can have a location without site_id and site_name

In model site_name is required

**Comment by drfaustusfade** [comment]
alright, let's try there were already such questions and requests for ET and NG for non-site reporting and optional site name, I think we can have that feature
please try to implement it

**Comment by fakhrihawari** [comment]
Mock up

https://user-images.githubusercontent.com/12655589/108474564-22eb4d00-72c2-11eb-9d71-6de449a222c4.mp4

**Comment by drfaustusfade** [comment]
Looks alright, what do you think also if visually having some kind of greyed out input, but on hover input would be active 

could you also please push commit, with not mandatory site name input for ET to have a test run

**Comment by drfaustusfade** [comment]
For engine and frontend, please confirm also that having  site name as non mandatory would not break anything

**Comment by fakhrihawari** [comment]
but i'm not done  NG coz NG have different form

**Comment by drfaustusfade** [comment]
Alright, we will try to migrate them later, when infinite scroll is done

**Comment by fakhrihawari** [comment]
ok

**Comment by fakhrihawari** [comment]
Testing for ET
 https://github.com/fakhrihawari/ngm-reportHub/tree/non-site_id-site_name
https://github.com/fakhrihawari/ngm-reportEngine/tree/non-site_id-site_name-be

Updated non site based old from too for NG

**Comment by drfaustusfade** [comment]
looks good,
_rename:_ No need to fill this => "-"
Site Name? => Site
SiteName => Site?
is required site_name should be removed on other collections in addition to TargetLocation? will that cause issues?

Note: as for now NG opted to still have required

let's have that feature for future use, but we will use just not mandatory site_name input for now so that not to confuse partners

- [ ] to check not required site_name input for ET without a checkbox

**Comment by fakhrihawari** [comment]
So for NG no need to change ?

**Comment by drfaustusfade** [comment]
no need for now, but they can decide later to have that feature

**Comment by fakhrihawari** [comment]
without checkbox
https://user-images.githubusercontent.com/12655589/108685233-7b228900-7526-11eb-89b9-54b45c24add4.mp4

If location do not have site_name it not appear in monthly report

**Comment by drfaustusfade** [comment]
what is the reason?

**Comment by fakhrihawari** [comment]
> what is the reason?

i dont know

**Comment by drfaustusfade** [comment]
alright, let's check, what are branches?

**Comment by fakhrihawari** [comment]
https://github.com/fakhrihawari/ngm-reportHub/tree/non-site_id-site_name
https://github.com/fakhrihawari/ngm-reportEngine/tree/non-site_id-site_name-be

that one

**Comment by fakhrihawari** [comment]
I found the problem, the site_name is set to undefined if its blank , i will fix it

**Comment by fakhrihawari** [comment]
Updated with test to check not required site_name input for ET without a checkbox and bug fix monthly report location without sitename

**Comment by drfaustusfade** [comment]
alright, would setting site_name defaultsTo "" fix also the problem? let's set it like that to safeguard. Also other collection required site_name would inherit from target_locations?

**Comment by drfaustusfade** [comment]
NG still have SiteName selection

**Comment by fakhrihawari** [comment]
> NG still have SiteName selection

**Comment by fakhrihawari** [comment]
> alright, would setting site_name defaultsTo "" fix also the problem? let's set it like that to safeguard. Also other collection required site_name would inherit from target_locations?

yes, i change the model for beneficairies and location for report like target location coz it will not appear in monthly report 

okay updated and remove hide sitename selection from NG

**Comment by drfaustusfade** [comment]
alright, some issues
that on adding location for ET, the site name is required 
and if removing the site name from target location, the site name will still be with old name

**Comment by fakhrihawari** [comment]
> alright, some issues
> that on adding location for ET, the site name is required
> and if removing the site name from target location, the site name will still be with old name

actually i already fix this before,  i comment the code that solve for that issue coz   i think  site_name defaultsTo "" will also fix that too  and  not to checki it again . But now it fix


https://user-images.githubusercontent.com/12655589/108792975-15c9a900-75b5-11eb-8815-a0c945a4b34c.mp4

**Comment by drfaustusfade** [comment]
ok, perfect, adding monthly report locations also updated?

> > alright, some issues
> > that on adding location for ET, the site name is required
> > and if removing the site name from target location, the site name will still be with old name
> 
> actually i already fix this before, i comment the code that solve for that issue coz i think site_name defaultsTo "" will also fix that too and not to checki it again . But now it fix
> 
>  fixbuget.mp4

**Comment by fakhrihawari** [comment]
> ok, perfect, adding monthly report locations also updated?
> 
> > > alright, some issues
> > > that on adding location for ET, the site name is required
> > > and if removing the site name from target location, the site name will still be with old name
> > 
> > 
> > actually i already fix this before, i comment the code that solve for that issue coz i think site_name defaultsTo "" will also fix that too and not to checki it again . But now it fix
> > fixbuget.mp4

updated for add location on monthly report location.  do we keep this "site_name_checked" atrribute?

**Comment by drfaustusfade** [comment]
i would propose to have it for the future us, in case we will need checkbox

**Comment by drfaustusfade** [comment]
```
// on monthly add location validate, I think !l.site_name is missing, otherwise errors everywhere except ET
if (l.admin0pcode !== 'ET') {
					id = "label[for='ngm-new_location-site_name']";
					$(id).addClass('error');
					validation.divs.push(id);
					complete = false;
				}
```

**Comment by fakhrihawari** [comment]
> ```
> // on monthly add location validate, I think !l.site_name is missing, otherwise errors everywhere except ET
> if (l.admin0pcode !== 'ET') {
> 					id = "label[for='ngm-new_location-site_name']";
> 					$(id).addClass('error');
> 					validation.divs.push(id);
> 					complete = false;
> 				}
> ```

Thank you ,  updated

**Comment by drfaustusfade** [comment]
```
if (d.admin0pcode === 'NG') {
								rowComplete++;
							}else{
								if (d.site_name) {
									rowComplete++;
								}
							}
```

NG have confirmed not required site_name yet, in that case, I think site_name should be checked

**Comment by drfaustusfade** [comment]
Location names on monthly reports without site name end as next
, null
, 
I think it is better visually not to have them

**Comment by fakhrihawari** [comment]
> ```
> if (d.admin0pcode === 'NG') {
> 								rowComplete++;
> 							}else{
> 								if (d.site_name) {
> 									rowComplete++;
> 								}
> 							}
> ```
> 
> NG have confirmed not required site_name yet, in that case, I think site_name should be checked

i already updated that too , for now that site_name not be checked only for ET

**Comment by fakhrihawari** [comment]
> Location names on monthly reports without site name end as next
> , null
> ,
> I think it is better visually not to have them

Updated, should we push?

**Comment by drfaustusfade** [comment]
looks alright, please recheck all functionality again where site_name is concerned and proceed with a pull request

**Comment by fakhrihawari** [comment]
okay

**Comment by drfaustusfade** [comment]
Notes:
markers look ok without site_name, i think no need to update

**Comment by fakhrihawari** [comment]
on PR

**Comment by drfaustusfade** [comment]
thank you, well done,
I was also thinking on forms, to have input next to checkbox like when site selection present with disabled input, or in place of checkbox to use selection from list, adding option to yes/no/ non-site

**Comment by fakhrihawari** [comment]
<img width="1149" alt="Screen Shot 2021-02-24 at 09 52 23" src="https://user-images.githubusercontent.com/12655589/108941781-3a428580-7688-11eb-8eb2-ab51fb652b86.png">
<img width="1149" alt="Screen Shot 2021-02-24 at 09 51 59" src="https://user-images.githubusercontent.com/12655589/108941784-3ca4df80-7688-11eb-911b-8eb94ae2f348.png">

link branch updated
https://github.com/fakhrihawari/ngm-reportHub/tree/non-site_id-site_name

is like that?

**Comment by drfaustusfade** [comment]
yes, could be, i think that is cleaner than the checkbox, by the cost of one more click, what do you think? 

other design option, as you did, would be in place of Site ? and From List ? selections to have checkboxes with same site name input on the same line rather than separate, but that would be hard to put in balance to other elements

Note:
to change Site Name ? to label just Site ?

**Comment by fakhrihawari** [comment]
Yes, it more aligned using select rather than checkbox

**Comment by fakhrihawari** [comment]
<img width="1183" alt="Screen Shot 2021-02-24 at 16 48 21" src="https://user-images.githubusercontent.com/12655589/108982301-37fc1d80-76c0-11eb-9124-1513eb1b33ff.png">

**Comment by drfaustusfade** [comment]
alright, lets have ? mark without space

**Comment by drfaustusfade** [comment]
NG confirmed not mandatory site name, please enable, but without _site?_ selection

**Comment by fakhrihawari** [comment]
> alright, lets have ? mark without space

updated

**Comment by fakhrihawari** [comment]
Link PR:
https://github.com/drfaustusfade/ngm-reportHub/pull/113

**Comment by drfaustusfade** [comment]
alright, there seem to be some bugs, please test and compile test cases conducted
also please split NG enabling non-mandatory input, in a separate commit
location form yes/no site selection into a separate commit
and for now not enable site yes/no selection for ET, 

I had
on adding location
![chrome_6gUXG6ORgC](https://user-images.githubusercontent.com/29349472/109132597-a351f800-7771-11eb-9318-1c429bf089d0.png)

on changing site yes/no
![chrome_11E1uhok8G](https://user-images.githubusercontent.com/29349472/109132986-18bdc880-7772-11eb-9f59-b18202e437bd.png)

also for some reason next
![chrome_xnT44EkB9N](https://user-images.githubusercontent.com/29349472/109133392-971a6a80-7772-11eb-9a1a-40f8e842bf9f.png)
after save 
![chrome_x2OxjWGu48](https://user-images.githubusercontent.com/29349472/109133383-94b81080-7772-11eb-874f-520f5e660c70.png)

**Comment by fakhrihawari** [comment]
Miss that one , i will try solve this and push it as soon as posible,
and "now not enable site yes/no selection for ET" i think on master already like that

**Comment by fakhrihawari** [comment]
Fix it like this

https://user-images.githubusercontent.com/12655589/109267733-a1a13680-783c-11eb-80df-8842232fdc7d.mp4

and sometime the site name not appear coz the type of site_id in location and in the list not same 
like below
<img width="1158" alt="Screen Shot 2021-02-26 at 14 36 24" src="https://user-images.githubusercontent.com/12655589/109270287-63a61180-7840-11eb-9b50-ace5c4a156bc.png">

is just on my local all in prod too?


link : https://github.com/fakhrihawari/ngm-reportHub/tree/non-site_id-site_name

if you okay, i will make separate commit for NG

**Comment by fakhrihawari** [comment]
NOTE: 
Some admin2,admin3 code in AdminSite is different from list Admin2List and admin3list
in admin2list
<img width="253" alt="Screen Shot 2021-03-01 at 10 18 09" src="https://user-images.githubusercontent.com/12655589/109447944-7b1b0f80-7a77-11eb-9e0c-b181690246a4.png">

in adminSIteList
<img width="279" alt="Screen Shot 2021-03-01 at 10 17 53" src="https://user-images.githubusercontent.com/12655589/109447972-879f6800-7a77-11eb-9bc2-d1fa01ceb005.png">
 

So if we select site_name from list it become like this
<img width="1051" alt="Screen Shot 2021-03-01 at 10 19 45" src="https://user-images.githubusercontent.com/12655589/109448080-bf0e1480-7a77-11eb-9e84-84b595335e41.png">

**Comment by fakhrihawari** [comment]
Link Hide Feature "Site?":
https://github.com/fakhrihawari/ngm-reportHub/tree/non-site_id-site_name
